src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -101,8 +103,18 @@ public Book create(@RequestBody Book book) {
     @PutMapping("/{id}")
     @ResponseStatus(HttpStatus.CREATED)
     public Book update(@RequestBody Book book, @PathVariable Long id) {
-        bookRepository.findById(id).orElseThrow(BookNotFoundException::new);
-        return bookRepository.save(book);
+        Book bookDB = bookRepository.findById(id).orElseThrow(BookNotFoundException::new);
+        bookDB
+                .genre(book.getGenre())
+                .author(book.getAuthor())
+                .image(book.getImage())
+                .title(book.getTitle())
+                .subtitle(book.getSubtitle())
+                .publisher(book.getPublisher())
+                .year(book.getYear())
+                .pages(book.getPages())
+                .isbn(book.getIsbn());
+        return bookRepository.save(bookDB);
```

Comment:
If I'm not wrong, you can do just this line without the setters of the properties, because inside the book parameter model comes the ID; so, if you save it like that it updates the book with the ID of the model.